File: examples/typescript/03-embeddings.ts
/**
 * Generating text embeddings via the TypeScript SDK.
 *
 * Run: pnpm embeddings
 */
import { SetuClient } from "@setu/sdk";

async function main() {
  const client = new SetuClient();

  const response = await client.embeddings.create({
    model: "text-embedding-3-small",
    input: ["Setu Gateway routes LLM requests.", "It also caches responses."],
  });

  for (const item of response.data as any[]) {
    console.log(`embedding[${item.index}]: dim=${item.embedding.length}, first values=${item.embedding.slice(0, 4)}`);
  }
}

main();

File: examples/python/03_embeddings.py
"""Generating text embeddings via the Python SDK.

Run: python 03_embeddings.py
"""

from setu_gateway import SetuClient


def main() -> None:
    client = SetuClient()

    response = client.embeddings.create(
        model="text-embedding-3-small",
        input=["Setu Gateway routes LLM requests.", "It also caches responses."],
    )

    for item in response["data"]:
        vec = item["embedding"]
        print(f"embedding[{item['index']}]: dim={len(vec)}, first values={vec[:4]}")


if __name__ == "__main__":
    main()
